compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -261,6 +266,221 @@ static CodeGenPipeline getTensorCorePipeline(Type elementType) {
   return codegenPipeline;
 }
 
+//====---------------------------------------------------------------------===//
+// Vector Distribution Reduction Pipeline Configuration
+//====---------------------------------------------------------------------===//
+
+/// Check if `op` is a linalg.reduce or a linalg.generic that has at least one
+/// reduction iterator.
+static bool hasReductionIterator(linalg::LinalgOp &op) {
+  return isa<linalg::ReduceOp, linalg::GenericOp>(op) &&
+         llvm::any_of(op.getIteratorTypesArray(), linalg::isReductionIterator);
+}
+
+// Get the bitwidth of the operation.
+static FailureOr<int64_t> getBitWidth(linalg::LinalgOp op) {
+
+  Value init = op.getDpsInitOperand(0)->get();
+  Value src = op.getDpsInputOperand(0)->get();
+  Type initElemType = getElementTypeOrSelf(init);
+  Type srcElemType = getElementTypeOrSelf(src);
+
+  if (auto initOp = init.getDefiningOp<linalg::GenericOp>()) {
+    if (IREE::LinalgExt::isBitExtendOp(initOp))
+      initElemType = getElementTypeOrSelf(initOp.getDpsInputs()[0]);
+  }
+
+  if (auto srcOp = src.getDefiningOp<linalg::GenericOp>()) {
+    if (IREE::LinalgExt::isBitExtendOp(srcOp))
+      srcElemType = getElementTypeOrSelf(srcOp.getDpsInputs()[0]);
+  }
+
+  if (!initElemType.isIntOrFloat() || !srcElemType.isIntOrFloat())
+    return failure();
+
+  int64_t bitWidth = std::min(initElemType.getIntOrFloatBitWidth(),
+                              srcElemType.getIntOrFloatBitWidth());
+
+  // Reduction distribution only supports 8/16/32 bit types now.
+  if (!llvm::is_contained({8, 16, 32}, bitWidth))
+    return failure();
+
+  return bitWidth;
+}
+
+/// The kernel config is for a single reduction op within a
+/// dispatch. The reduction op shouldn't have any consumer
+/// because that may introduce new dimensions and the
+/// vector distribution analysis might fail. The lowering config is
+/// only attached to the reduction op for now.
+/// TODO(pashu123): Analyze the shared parallel and reduction
+/// dimensions within a dispatch. Based on the analysis try to
+/// put the lowering_config on each of the operation within a
+/// dispatch. That way we can support multiple reduction ops
+/// within a dispatch.
+static LogicalResult
+setReductionVectorDistributionConfig(IREE::GPU::TargetAttr target,
+                                     mlir::FunctionOpInterface entryPoint,
+                                     linalg::LinalgOp op) {
+
+  int64_t countReduction = 0;
+
+  // TODO(pashu123): Remove this check and allow multiple reductions in a single
+  // dispatch.
+  WalkResult walkResult = entryPoint.walk([&](Operation *op) {
+    if (auto linalgOp = dyn_cast<linalg::LinalgOp>(op)) {
+      if (hasReductionIterator(linalgOp)) {
+        countReduction++;
```

Comment:
```suggestion
        ++countReduction;
```
See https://llvm.org/docs/CodingStandards.html#prefer-preincrement